### PR TITLE
Scaladoc on AnyRef#wait

### PR DESCRIPTION
**Pull request by eed3si9n**
Ref https://github.com/scala/bug/issues/11310

As a background, AnyRef is a fictional class for documentation only. This is achieved by a special mode in scaladoc called `-doc-no-compile`, which accepts `src/library-aux`.

Internally, AnyRef is a type alias of Object (for example in definitions.AnyRefClass).
Object has 3 overloads of wait method, but `Uncompilable` mode was matching on the method name so it was returning a field named `wait` to be paired with the Scaladocs on `AnyRef#wait` methods.

This fixes that by checking both the name and the arity for `scala.AnyRef#wait`. Here's from running `doc` after switching starr to locally published 2.13.3-pre-271afd8.

wait

### CODE REVIEW

**Comment by smarter** [comment]
The documentation for these methods in AnyRef.scala is dangerously wrong:

> Causes the current Thread to wait until another Thread invokes, the notify() or notifyAll() methods, or a specified amount of time has elapsed.

This ignores the phenomenon of spurious wakeups, I'd rather not display anything than display this. Best of all would be to replace that documentation by a link to the corresponding method in https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html which is actually comprehensive.

**Comment by eed3si9n** [comment]
@smarter I addressed it in 69a259a, but maybe whether that's the right direction to go should be debated in scala/bug#11310 or elsewhere?

File: src/scaladoc/scala/tools/nsc/doc/Uncompilable.scala [hunk 1]
```diff
@@ -23,22 +23,39 @@ trait Uncompilable {
...
+    p.nameChain match {
+      // for scala.AnyRef.wait, pick Object#wait with the same arity
+      case xs if xs.map(_.toString) == WaitNames =>
+        val arity = p.docDef.definition match {
+          case t: DocParser#DefDef => t.vparamss.flatten.size
+          case _ => 0
+        }
+        val wait = AnyRefTpe.members find {
+          case m: MethodSymbol =>
+            m.name == newTermName("wait") && m.paramLists.flatten.size == arity
```
Comment (lrytz) [inline src/scaladoc/scala/tools/nsc/doc/Uncompilable.scala:45]: maybe still add a `case _ => false` (even if members of `AnyRef` are unlikely to change)

**Review by lrytz — APPROVED** [review]

File: src/scaladoc/scala/tools/nsc/doc/Uncompilable.scala (hunk 1 above)
Comment (eed3si9n) [reply src/scaladoc/scala/tools/nsc/doc/Uncompilable.scala:45]: Addressed in 9f685f78f5373cd98c6ea22873402ee613de1542

Comment (lrytz) [reply src/scaladoc/scala/tools/nsc/doc/Uncompilable.scala:45]: 👍 thanks!

**Pull request merged**